fili-core/src/test/groovy/com/yahoo/bard/webservice/data/DruidResponseParserSpec.groovy

```diff
@@ -127,17 +127,17 @@ class DruidResponseParserSpec extends Specification {
 
         and:
         Result result = resultSet.get(0)
-        result.getDimensionRow(schema.columns.toArray()[0])?.get(BardDimensionField.DESC) == "4"
-        result.getDimensionRow(schema.columns.toArray()[1])?.get(BardDimensionField.DESC) == "u"
-        result.getDimensionRow(schema.columns.toArray()[2])?.get(BardDimensionField.DESC) == ""
-        result.getDimensionRow(schema.columns.toArray()[2])?.get(BardDimensionField.ID) == "foo"
-        resultSet[0].getMetricValueAsNumber(schema.getColumn("pageViews")) == 1 as BigDecimal
-        resultSet[0].getMetricValueAsNumber(schema.getColumn("time_spent")) == 2 as BigDecimal
+        result.getDimensionRow((DimensionColumn) schema.columns.toArray()[0])?.get(BardDimensionField.DESC) == "u"
+        result.getDimensionRow((DimensionColumn) schema.columns.toArray()[2])?.get(BardDimensionField.DESC) == "4"
+        result.getDimensionRow((DimensionColumn) schema.columns.toArray()[3])?.get(BardDimensionField.DESC) == ""
+        result.getDimensionRow((DimensionColumn) schema.columns.toArray()[3])?.get(BardDimensionField.ID) == "foo"
+        resultSet[0].getMetricValueAsNumber(schema.getColumn("pageViews", MetricColumn.class).get()) == 1 as BigDecimal
+        resultSet[0].getMetricValueAsNumber(schema.getColumn("time_spent", MetricColumn.class).get()) == 2 as BigDecimal
```

Comment:
The order of things seems to changed, just trying to fix errors, we need to write better code than this which relies on ordering.